Remotion/Validation/UnitTests/Implementation/FallbackValidationMessageFactoryTest.cs

```diff
@@ -15,23 +15,84 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using Moq;
 using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.UnitTests.TestDomain;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Implementation
 {
   [TestFixture]
   public class FallbackValidationMessageFactoryTest
   {
     [Test]
-    [Ignore("RM-5906")]
     public void CreateValidationMessageForPropertyValidator ()
     {
+      var propertyValidatorMock = new Mock<IPropertyValidator>();
+      var validatedPropertyMock = new Mock<IPropertyInformation>();
+
+      var fallbackFactory = new FallbackValidationMessageFactory();
+      var result = fallbackFactory.CreateValidationMessageForPropertyValidator(propertyValidatorMock.Object, validatedPropertyMock.Object);
+
+      Assert.That(result, Is.Not.Null);
+      Assert.That(result, Is.InstanceOf<InvariantValidationMessage>());
+      Assert.That(result.ToString(), Is.EqualTo($"IPropertyValidatorProxy: Validation error."));
+    }
+
+    [Test]
+    public void CreateValidationMessageForPropertyValidator_WithNullAsValidator_ThrowsArgumentNullException ()
+    {
+      var validatedPropertyMock = new Mock<IPropertyInformation>();
+
+      var fallbackFactory = new FallbackValidationMessageFactory();
+
+      Assert.That(() => fallbackFactory.CreateValidationMessageForPropertyValidator(null, validatedPropertyMock.Object), Throws.ArgumentNullException);
+    }
```

Comment:
Typically, we don't test argument-null and argument-empty exceptions because our infrastructre provides the helpers, they are everywhere and it's just a matter of forgetting the call. Since we don't have them in other places, I would recommend dropping them. I would also place in APIs where the signature allows null values but we expcicitly disallow null. This would be done to document/enforce the API rules and not to unit test that we haven't made a mistake during implementation.

As for future stuff: there was a long time hoping that Microsoft would solve arg-null-guards via the compiler. Since they dropped this again, it could be a simple option to use roslyn to verify that each parameter has an argument null check.